tests/Collection/ArrayCollectionHelperTest.php

```diff
@@ -0,0 +1,363 @@
+                'debug',
+            ],
+            'foo',
+        ];
+
+        $result = ArrayCollectionHelper::merge($a, $b, $c)->toArray();
+        $expected = [
+            'name' => 'Yii',
+            'version' => '2.0',
+            'options' => [
+                'namespace' => true,
+                'unittest' => true,
+            ],
+            'features' => [
+                'mvc',
+                'gii',
+                'debug',
+            ],
+            'foo',
+        ];
+
+        $this->assertEquals($expected, $result);
+    }
+
+    public function testMergeWithUnset(): void
+    {
+        $a = [
+            'name' => 'Yii',
+            'version' => '1.0',
+            'options' => [
+                'namespace' => false,
+                'unittest' => false,
+            ],
+            'features' => [
+                'mvc',
+            ],
+        ];
+        $b = (new ArrayCollection([
+            'version' => '1.1',
+            // 'options' => new UnsetValue(),
+            'features' => [
+                'gii',
+            ],
+        ]))->addModifier(
+            (new UnsetValue())->forKey('options')
+        );
+
+        $result = ArrayCollectionHelper::merge($a, $b)->toArray();
+        $expected = [
+            'name' => 'Yii',
+            'version' => '1.1',
+            'features' => [
+                'mvc',
+                'gii',
+            ],
+        ];
+
+        $this->assertEquals($expected, $result);
+    }
+
+    public function testMergeWithReplace(): void
+    {
+        $a = [
+            'name' => 'Yii',
+            'version' => '1.0',
+            'options' => [
+                'namespace' => false,
+                'unittest' => false,
+            ],
+            'features' => [
+                'mvc',
+            ],
+        ];
+        $b = (new ArrayCollection([
+            'version' => '1.1',
+            'options' => [
+                'unittest' => true,
+            ],
+//            'features' => new ReplaceValue(
+//                [
+//                    'gii',
+//                ]
+//            ),
+        ]))->addModifier(
+            (new ReplaceValue())
+                ->forKey('features')
+                ->toValue(['gii'])
+        );
+
+        $result = ArrayCollectionHelper::merge($a, $b)->toArray();
+        $expected = [
+            'name' => 'Yii',
+            'version' => '1.1',
+            'options' => [
+                'namespace' => false,
+                'unittest' => true,
+            ],
+            'features' => [
+                'gii',
+            ],
+        ];
+
+        $this->assertEquals($expected, $result);
+    }
+
+    public function testMergeWithRemoveKeys(): void
+    {
+        $a = [
+            'name' => 'Yii',
+            'version' => '1.0',
+        ];
+        $b = (new ArrayCollection([
+            'version' => '1.1',
+            'options' => [],
+//            new RemoveKeys(),
```

Comment:
?

Reply:
This file with tests is copy of current array modifiers test adpated to collections.
I commented old modifiers for comparsions was / became.